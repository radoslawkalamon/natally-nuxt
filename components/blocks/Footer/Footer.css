.footer {
  align-items: center;
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);
}
