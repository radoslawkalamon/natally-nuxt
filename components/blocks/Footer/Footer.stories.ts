import type { Meta, Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import BlockFooter from '@/components/blocks/Footer/Footer.vue'

const meta: Meta = {
  title: 'Blocks / Footer',
  component: BlockFooter
}
export default meta

export const Default: Story = () => ({
  beforeMount () {
    // @ts-ignore
    this.$root.$on('privacy/modal/toggle', action('privacy/modal/toggle'))
  },
  components: {
    BlockFooter
  },
  template: '<BlockFooter />'
})
