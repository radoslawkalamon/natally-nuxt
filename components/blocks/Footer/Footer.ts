import Vue from 'vue'
import pkg from '@/package.json'

export default Vue.extend({
  name: 'BlocksFooter',
  components: {
    ComponentsLogoCrustyBread: () => import(/* webpackChunkName: "components-logo-crusty-bread" */'@/components/components/LogoCrustyBread/LogoCrustyBread.vue'),
    ComponentsNavigationFooter: () => import(/* webpackChunkName: "components-navigation-footer" */'@/components/components/NavigationFooter/NavigationFooter.vue'),
    WrappersText: () => import(/* webpackChunkName: "wrappers" */'@/components/wrappers/Text/Text.vue')
  },
  computed: {
    crustyBreadTechWebsite (): string {
      return 'https://pieczywotwarde.pl'
    },
    fullYear (): string {
      return `${(new Date()).getFullYear()}`
    },
    websiteName (): string {
      return '169cm.pl'
    },
    commitSha (): string {
      return process.env.DEPLOY_COMMIT_SHA || `v${pkg.version}`
    }
  }
})
