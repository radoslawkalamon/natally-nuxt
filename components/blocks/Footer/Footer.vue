<template>
  <footer class="footer">
    <ComponentsNavigationFooter />
    <a
      :href="crustyBreadTechWebsite"
      target="_blank"
      rel="noopener noreferrer"
    >
      <ComponentsLogoCrustyBread />
    </a>
    <WrappersText>
      <p>{{ websiteName }} &copy; {{ fullYear }}</p>
    </WrappersText>
    <WrappersText class="footer__commit-sha">
      <p>{{ commitSha }}</p>
    </WrappersText>
  </footer>
</template>

<script src="./Footer.ts" lang="ts"></script>
<style src="./Footer.css" scoped></style>
