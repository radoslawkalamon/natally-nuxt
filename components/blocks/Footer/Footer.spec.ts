import Footer from '@/components/blocks/Footer/Footer.vue'
import { shallPassIntegrationSanityTest, shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

jest.mock('@/package.json', () => ({ version: 'TEST_ENVIROMENT' }))
jest
  .useFakeTimers()
  .setSystemTime(new Date('2021-11-01').getTime())

describe('Blocks / Footer', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: Footer,
      options: {
        stubs: [
          'ComponentsLogoCrustyBread',
          'ComponentsNavigationFooter',
          'WrappersText'
        ]
      }
    })
  })

  describe('Integration', () => {
    shallPassIntegrationSanityTest({ component: Footer })
  })
})
