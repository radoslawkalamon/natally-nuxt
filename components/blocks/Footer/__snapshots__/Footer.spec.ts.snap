// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / Footer Integration shall pass integration sanity test 1`] = `
"<footer class=\\"footer\\">
  <nav class=\\"navigation-footer\\">
    <ul class=\\"navigation\\">
      <li class=\\"navigation__item\\">
        <nuxtlink to=\\"/polityka-prywatnosci/\\" class=\\"link link--general\\">Polityka prywatności</nuxtlink>
      </li>
      <li class=\\"navigation__item\\"><button type=\\"button\\" data-test=\\"open-privacy-modal\\" class=\\"button button--text\\">Ustawienia prywatności</button></li>
    </ul>
  </nav> <a href=\\"https://pieczywotwarde.pl\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><img alt=\\"Logo firmy Pieczywo Twarde\\" height=\\"128\\" src=\\"./images/crusty-bread.svg\\" title=\\"Logo firmy Pieczywo Twarde\\" width=\\"56\\"></a>
  <div class=\\"text\\">
    <p>169cm.pl © 2021</p>
  </div>
  <div class=\\"footer__commit-sha text\\">
    <p>vTEST_ENVIROMENT</p>
  </div>
</footer>"
`;

exports[`Blocks / Footer Unit shall pass unit sanity test 1`] = `
"<footer class=\\"footer\\">
  <componentsnavigationfooter-stub></componentsnavigationfooter-stub> <a href=\\"https://pieczywotwarde.pl\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
    <componentslogocrustybread-stub></componentslogocrustybread-stub>
  </a>
  <wrapperstext-stub>
    <p>169cm.pl © 2021</p>
  </wrapperstext-stub>
  <wrapperstext-stub class=\\"footer__commit-sha\\">
    <p>vTEST_ENVIROMENT</p>
  </wrapperstext-stub>
</footer>"
`;
